libs/kml/color_parser.cpp

```diff
@@ -0,0 +1,122 @@
+#include "color_parser.hpp"
+
+#include "coding/hex.hpp"
+
+#include "base/string_utils.hpp"
+
+namespace kml
+{
+
+struct RGBColor
+{
+  uint8_t r, g, b;
+};
+
+std::optional<uint32_t> ParseHexColor(std::string_view c)
+{
+  if (c.empty())
+    return {};
+
+  if (c.front() == '#')
+    c.remove_prefix(1);
+  if (c.size() != 6 && c.size() != 8)
+    return {};
+
+  auto const colorBytes = FromHex(c);
+  switch (colorBytes.size())
+  {
+  case 3: return ToRGBA(colorBytes[0], colorBytes[1], colorBytes[2]);
+  case 4: return ToRGBA(colorBytes[1], colorBytes[2], colorBytes[3], colorBytes[0]);
+  default: return {};
+  }
+}
+
+// Garmin extensions spec: https://www8.garmin.com/xmlschemas/GpxExtensionsv3.xsd
+// Color mapping: https://help.locusmap.eu/topic/extend-garmin-gpx-compatibilty
+std::optional<uint32_t> ParseGarminColor(std::string_view c)
+{
+  /// @todo Unify with RGBColor instead of string.
+  static std::pair<std::string_view, std::string_view> arrColors[] = {
```

Comment:
nit: Non-const static function member generates too much overhead in the assembly code. Move to static const/constexpr global variable instead.